### PR TITLE
EBS-149 Textarea component improvements

### PR DESCRIPTION
**Pull request by a7madgamal**
breaking changes to `Textarea`
removed: 
- `onInput` since `onChange` is clearer and matches most other components
- `required` is now optional to match `InputWrapper` usage, I think we should change it later to `required` since it's weird to mark optional fields with `*` IMO.

added: 
- `id`, `name`, `defaultValue`, `iconName`, `onChange`, `onBlur`, `validateOnChange` and `validateOnBlur`.

now uses the `ValidationContext` and InputWrapper.

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,8 +13,8 @@
 /public/dist/*.js.map
 
 # dist files
-/dist/*
-/storybook-static/*
+/dist
```
Comment (a7madgamal) [inline .gitignore:16]: just to color ignored folders in vscode properly :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
       - run: npm run linter:check
       - run:
           name: Tests
-          no_output_timeout: 15m # a workaround until we make it faster later
+          no_output_timeout: 30m # a workaround until we make it faster later
```
Comment (a7madgamal) [inline .circleci/config.yml:41]: the time increases with each new story 🤷

File: package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "storybook:build": "build-storybook -s public",
     "build:package": "babel --delete-dir-on-start --extensions .ts,.tsx,.js,.jsx ./src --out-dir ./dist && tsc --project tsconfig.types.json --emitDeclarationOnly",
     "test": "npm run build:package && jest",
-    "test:update": "jest --updateSnapshot",
+    "test:update": "npm run build:package && npm run storybook:build && jest --updateSnapshot",
```
Comment (a7madgamal) [inline package.json:25]: this makes sure everything is ready before snapshots

Comment (Weetbix) [reply package.json:25]: nice one, thanks for fixing

File: tests/helpers.js [hunk 4]
```diff
@@ -1,16 +1,11 @@
...
+  const location = path.join(__dirname, "..", "storybook-static");
+  console.log({ location });
 
-  if (fs.existsSync(location)) {
-    location = `file://${location}`;
-  } else {
-    location = "http://localhost:3006";
-  }
-  return location;
+  return `file://${location}`;
```
Comment (Weetbix) [inline tests/helpers.js:8]: I think this change makes sense. We should only support one method of snapshotting that is identical everywhere

File: tests/helpers.js [hunk 5]
```diff
@@ -1,16 +1,11 @@
 import path from "path";
-import fs from "fs";
 import devices from "puppeteer/DeviceDescriptors";
 
 export function getStorybookUrl() {
-  let location = path.join(__dirname, "..", "storybook-static");
+  const location = path.join(__dirname, "..", "storybook-static");
+  console.log({ location });
```
Comment (Weetbix) [inline tests/helpers.js:6]: Is this intentional?

Comment (a7madgamal) [reply tests/helpers.js:6]: lol nope

**Comment by a7madgamal** [comment]
@Weetbix I check the css seems similar now

**Pull request merged**